process/smartContract/scQueryService.go

```diff
@@ -42,26 +44,57 @@ func NewSCQueryService(
 
 // ExecuteQuery returns the VMOutput resulted upon running the function on the smart contract
 func (service *SCQueryService) ExecuteQuery(query *process.SCQuery) (*vmcommon.VMOutput, error) {
-	if query.ScAddress == nil {
-		return nil, process.ErrNilScAddress
+	err := checkQueryFields(query)
```

Comment:
No need for second function. Use ExecuteQuery and fill the value and callerAddress as they were before if those are empty.

Reply:
done